projects/opendr_ws_2/src/opendr_perception/README.md

```diff
@@ -0,0 +1,874 @@
+   - `--device DEVICE`: device to use, either `cpu` or `cuda`, falls back to `cpu` if GPU or CUDA is not found (default=`cuda`)
+   - `-td --temp_dir TEMP_DIR`: path to a temporary directory with models (default=`temp`)
+
+3. Default output topics:
+   - Output images: `/opendr/image_objects_annotated`
+   - Detection messages: `/opendr/objects`
+   - Tracking ID messages: `/opendr/objects_tracking_id`
+
+   For viewing the output, refer to the [notes above.](#notes)
+
+**Notes**
+
+An [image dataset node](#image-dataset-ros2-node) is also provided to be used along these nodes.
+Make sure to change the default input topic of the tracking node if you are not using the USB cam node.
+
+### Panoptic Segmentation ROS2 Node
+
+You can find the panoptic segmentation ROS2 node python script [here](./opendr_perception/panoptic_segmentation_efficient_ps_node.py) to inspect the code and modify it as you wish to fit your needs.
+The node makes use of the toolkit's [panoptic segmentation tool](../../../../src/opendr/perception/panoptic_segmentation/efficient_ps/efficient_ps_learner.py) whose documentation can be found [here](../../../../docs/reference/efficient-ps.md)
+and additional information about Efficient PS [here](../../../../src/opendr/perception/panoptic_segmentation/README.md).
+
+#### Instructions for basic usage:
+
+1. Start the node responsible for publishing images. If you have a USB camera, then you can use the `usb_cam_node` as explained in the [prerequisites above](#prerequisites).
+
+2. You are then ready to start the panoptic segmentation node:
+
+    ```shell
+    ros2 run opendr_perception panoptic_segmentation_efficient_ps_node.py
+    ```
+
+    The following optional arguments are available:
+   - `-h or --help`: show a help message and exit
+   - `-i or --input_rgb_image_topic INPUT_RGB_IMAGE_TOPIC` : listen to RGB images on this topic (default=`/image_raw`)
+   - `-oh --output_heatmap_topic OUTPUT_HEATMAP_TOPIC`: publish the semantic and instance maps on this topic as `OUTPUT_HEATMAP_TOPIC/semantic` and `OUTPUT_HEATMAP_TOPIC/instance`, `None` to stop the node from publishing on this topic (default=`/opendr/panoptic`)
+   - `-ov --output_rgb_image_topic OUTPUT_RGB_IMAGE_TOPIC`: publish the panoptic segmentation map as an RGB image on this topic or a more detailed overview if using the `--detailed_visualization` flag, `None` to stop the node from publishing on this topic (default=`opendr/panoptic/rgb_visualization`)
+   - `--detailed_visualization`: generate a combined overview of the input RGB image and the semantic, instance, and panoptic segmentation maps and publish it on `OUTPUT_RGB_IMAGE_TOPIC` (default=deactivated)
+   - `--checkpoint CHECKPOINT` : download pretrained models [cityscapes, kitti] or load from the provided path (default=`cityscapes`)
+
+3. Default output topics:
+   - Output images: `/opendr/panoptic/semantic`, `/opendr/panoptic/instance`, `/opendr/panoptic/rgb_visualization`
```

Comment:
for `rgb_visualization` I got this error: `Failed to generate prediction: to_ros_image() got an unexpected keyword argument 'encoder'`